thirdeye-ui/src/app/components/subscription-group-view/alert-associations-view-table/alert-associations-view-table.component.tsx

```diff
@@ -0,0 +1,112 @@
+ * either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import { Box, Typography } from "@material-ui/core";
+import { capitalize, sortBy } from "lodash";
+import React, { FunctionComponent, ReactNode, useCallback } from "react";
+import { useTranslation } from "react-i18next";
+import { DataGridColumnV1, DataGridV1 } from "../../../platform/components";
+import { DataGridV1Props } from "../../../platform/components/data-grid-v1/data-grid-v1/data-grid-v1.interfaces";
+import { linkRendererV1 } from "../../../platform/utils";
+import { getAlertsAlertPath } from "../../../utils/routes/routes.util";
+import { EmptyStateSwitch } from "../../page-states/empty-state-switch/empty-state-switch.component";
+import {
+    AlertAssociationsViewTableProps,
+    UiAssociation,
+} from "./alert-associations-view-table.interfaces";
+import { getUiAssociation } from "./alert-associations-view-table.utils";
+
+export const AlertAssociationsViewTable: FunctionComponent<AlertAssociationsViewTableProps> =
+    ({ uiSubscriptionGroup, customToolbar }) => {
+        const { t } = useTranslation();
+        const uiAssociations = getUiAssociation(
+            uiSubscriptionGroup.alerts || [],
+            t
+        );
+
+        const alertNameRenderer = useCallback(
+            (
+                cellValue: Record<string, unknown>,
+                data: UiAssociation
+            ): ReactNode =>
+                linkRendererV1(
+                    cellValue,
+                    getAlertsAlertPath(data.alertId),
+                    false,
+                    `${t("label.view")} ${t("label.alert")}:${cellValue}`
+                ),
+            []
+        );
+
+        const dataGridColumns: DataGridColumnV1<UiAssociation>[] = [
+            {
+                key: "alertName",
+                dataKey: "alertName",
+                header: t("label.alert-name"),
+                minWidth: 150,
+                flex: 1,
+                customCellRenderer: alertNameRenderer,
+            },
+            {
+                key: "enumerationName",
+                dataKey: "enumerationName",
+                header: t("label.dimensions"),
+                minWidth: 150,
+                flex: 1,
+            },
+        ];
+
+        const dataGridProps: DataGridV1Props<UiAssociation> = {
+            data: sortBy(uiAssociations, "id"),
+            columns: dataGridColumns,
+            rowKey: "id",
+            hideBorder: true,
+            disableSelection: true,
+            toolbarComponent: customToolbar,
+        };
+
+        const isTableEmpty = uiAssociations.length === 0;
+
+        return (
+            <>
+                <Box height={isTableEmpty ? 100 : 400}>
+                    <DataGridV1<UiAssociation> {...dataGridProps} />
+                </Box>
+                <EmptyStateSwitch emptyState={null} isEmpty={!isTableEmpty}>
```

Comment:
we want to use `!isTableEmpty`?

Reply:
Yep. Since we are showing the toolbar and column headers even with empty data.

![Screenshot_145](https://user-images.githubusercontent.com/20799363/220959385-7f512466-6f02-4547-8e48-4d56a533e71c.png)

Reply:
But on second thoughts, it might be looking better _without_ the data grid components being rendered in the empty state.

![Screenshot_146](https://user-images.githubusercontent.com/20799363/220959509-ae97dabc-f8cc-464f-b3c5-56f1fe883bbd.png)

Reply:
I'll change it to the latter

Reply:
Updated. Looks better now.

![Screenshot_147](https://user-images.githubusercontent.com/20799363/220966801-d573787f-e7ed-4068-a486-271291de48f2.png)

![Screenshot_148](https://user-images.githubusercontent.com/20799363/220966838-9ac911d0-1ec1-43d7-ab03-14f85c9d3284.png)

Reply:
Use the empty Wine glass icon

Reply:
Otherwise looks good with the call to action

Reply:
I got this icon from the [UX mocks](https://www.figma.com/file/0D0Pbhs5cMtT55ticaZbf1/Subscription-group-%2F-Re-design-%2F-Subscription-groups-interaction-redesign?node-id=558%3A41844&t=I0lTPIjR9KvqKCDV-0). Should I prefer the wine glass icon instead?

![Screenshot_152](https://user-images.githubusercontent.com/20799363/220975834-4726c869-2d8b-40b2-a832-48a352292bfa.png)

Reply:
hmm ok let's stick with whats in the mocks